pkg/translator/prometheus/normalize_name_test.go

```diff
@@ -146,6 +149,45 @@ func TestOtelReceivers(t *testing.T) {
 
 }
 
+func TestTrimPromSuffixes(t *testing.T) {
+	registry := featuregate.NewRegistry()
+
+	require.NoError(t, registry.RegisterID(normalizeNameGateID, featuregate.StageBeta))
+
+	normalizer := NewNormalizer(registry)
+
+	assert.Equal(t, "active_directory_ds_replication_network_io", normalizer.TrimPromSuffixes("active_directory_ds_replication_network_io_bytes_total", pmetric.MetricTypeSum, "By"))
```

Comment:
As explained above, these units are not valid per OpenMetrics: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#unit

`Unit specifies MetricFamily units. If non-empty, it MUST be a suffix of the MetricFamily name separated by an underscore. Be aware that further generation rules might make it an infix in the text format.`

The unit on this metric should be `bytes`

Reply:
I've remove the unit conversion, and kept few tests cases with mismatching units and suffix to make sure these are not trimmed accidentally.